### ISSUE TITLE
Slack notifications not working

### ISSUE THREAD

**Issue by rahul799** [argoproj-labs/argocd-notifications#406]
## Summary 

I followed all the steps mentioned [here](https://argocd-notifications.readthedocs.io/en/stable/services/slack/) with an **on-sync** subscription method. But whenever sync is going on I can see below logs 
`level=info msg="Trigger on-sync-running result: []" app=argocd/grafana`
in `argocd-notifications-controlle` pod.
There are no notifications on slack.

I am using KOPS to manage my cluster.

I am using latest versions of argocd and notification controller

```
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/kafka-exporter
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/kafka-exporter
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/website
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/website
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/rumbl-admin
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/rumbl-admin
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/metric-server
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/metric-server
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/kubernetes-node-exporter
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/kubernetes-node-exporter
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/dhaniya-server
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/dhaniya-server
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/kube-state-metrics
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/kube-state-metrics
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/rumbl-js
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/rumbl-js
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/prometheus
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/prometheus
time="2022-06-06T11:51:52Z" level=info msg="Start processing" app=argocd/grafana
time="2022-06-06T11:51:52Z" level=info msg="Trigger on-sync-succeeded result: []" app=argocd/grafana
time="2022-06-06T11:51:52Z" level=info msg="Trigger on-sync-running result: []" app=argocd/grafana
time="2022-06-06T11:51:52Z" level=info msg="Processing completed" app=argocd/grafana
time="2022-06-06T11:52:23Z" level=info msg="Settings had been updated. Restarting controller..."
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by Geethree** [comment]
This has been hitting me as well. I've also noticed that the [troubleshooting for the argocd-notification controller deployed as a part of argocd](https://argo-cd.readthedocs.io/en/stable/operator-manual/notifications/troubleshooting/#in-your-cluster) doesn't  work.

It appears as if the subcommands aren't there as it tries to start argocd-notifications... help only shows flags

```
gerald@gerald-laptop:~/playground/spinnaker-cloud-api-pipelines$ kubectl -n argocd exec -it argocd-notifications-controller-56748c87bd-22rcs -- /usr/local/bin/argocd-notifications trigger get
INFO[0000] serving metrics on port 9001                 
INFO[0000] loading configuration 9001                   
FATA[0000] listen tcp 0.0.0.0:9001: bind: address already in use 
command terminated with exit code 1
```

There is only one pid in the pod.. argocd-notifications pid 1.

**Comment by seonggwonyoon** [comment]
I'm having this problem, too

**Comment by anfallow** [comment]
Hello, any updates or estimates on this fix? 

Thanks

**Comment by hebestreit** [comment]
This project has moved to the main Argo CD repository which was announced in the [README.md](https://github.com/argoproj-labs/argocd-notifications#argo-cd-notifications-is-now-part-of-argo-cd). 

To test the notification you now have to use following command which will print out details on success or failure.
```
argocd admin notifications template notify NAME RESOURCE_NAME [flags]
```

**Examples**

```
# Trigger notification using in-cluster config map and secret
argocd admin notifications template notify app-sync-succeeded guestbook --recipient slack:my-slack-channel

# Render notification render generated notification in console
argocd admin notifications template notify app-sync-succeeded guestbook
```

Find more information in the official documentation:

https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_admin_notifications_template_notify/

Also make sure that the template you're defining as `NAME` exists in the `argocd/argocd-notifications-cm` ConfigMap.

**Comment by Geethree** [comment]
Ok thanks @hebestreit Glad the documentation is updated now ;)

This helped me realize I didn't have any triggers or templates set.

TL;DR 
I modified my argocd-notifications-cm to be
````yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-notifications-cm
data:
  service.slack: |
    token: $slack-token
  defaultTriggers: |
    - on-deployed
  trigger.on-deployed: |
    - description: Application is synced and healthy. Triggered once per commit.
      oncePer: app.status.sync.revision
      send:
      - app-deployed
      when: app.status.operationState.phase in ['Succeeded'] and app.status.health.status == 'Healthy'
  template.app-deployed: |
    message: |
      {{if eq .serviceType "slack"}}:white_check_mark:{{end}} Application {{.app.metadata.name}} is now running new version of deployments manifests.
    slack:
      attachments: |
        [{
          "title": "{{ .app.metadata.name}}",
          "title_link":"{{.context.argocdUrl}}/applications/{{.app.metadata.name}}",
          "color": "#18be52",
          "fields": [
          {
            "title": "Sync Status",
            "value": "{{.app.status.sync.status}}",
            "short": true
          },
          {
            "title": "Repository",
            "value": "{{.app.spec.source.repoURL}}",
            "short": true
          },
          {
            "title": "Revision",
            "value": "{{.app.status.sync.revision}}",
            "short": true
          }
          {{range $index, $c := .app.status.conditions}}
          {{if not $index}},{{end}}
          {{if $index}},{{end}}
          {
            "title": "{{$c.type}}",
            "value": "{{$c.message}}",
            "short": true
          }
          {{end}}
          ]
        }]
        ```


And I have notifications and the `argocd admin notifications` tooling works =)

**Comment by deybhayden** [comment]
Just going to add that I had the same issue as GeeThree - I didn't realize that there were no default Triggers & Templates being shipped with the Notification Controller. Once I updated my helm install with the commented defaults starting [here](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L2368) things started working as expected. 👍🏻 I think this issue could be closed.
